Add metallb to docs (#2558)
aws/eks-anywhere: docs/content/en/docs/reference/packagespec/metallb/v0.12.1.md

---
title: "v0.12.1"
linkTitle: "v0.12.1"
weight: 20
description: >
---

{{% pageinfo %}}
[FRRouting](https://frrouting.org/) is currently not supported for MetalLB.
{{% /pageinfo %}}

### Configuring MetalLB in EKS Anywhere package spec

| Parameter | Description | Default |
|---|---|---|
|**General**|||

> | address-pools[] | List of **address-pool** objects. Address-pools list the IP addresses that MetalLB is allowed to allocate, along with settings for how to advertise those addresses over BGP once assigned. You can have as many address pools as you want<br/>Example:<br/><pre>address-pools:<br/>  - name: default<br/>    protocol: bgp<br/>    addresses:<br/>      - 10.220.0.90/30| None |

Reviewer: ```suggestion
| address-pools[] | List of **address-pool** objects. Address-pools list the IP addresses that MetalLB is allowed to allocate, along with settings for how to advertise those addresses over BGP once assigned. You can have as many address pools as you want.<br/>Example:<br/><pre>address-pools:<br/>  - name: default<br/>    protocol: bgp<br/>    addresses:<br/>      - 10.220.0.90/30| None |
```

> | peers[] | List of **peer** objects. The peers list tells MetalLB what BGP routers to connect too. There is one entry for each router you want to peer with. <br/>Example:<br/><pre>peers:<br/>  - peer-address: 10.220.0.2<br/>    peer-asn: 65000<br/>    my-asn: 65002</pre> | |

Reviewer: ```suggestion
| peers[] | List of **peer** objects. The peers list tells MetalLB what BGP routers to connect to. There is one entry for each router you want to peer with. <br/>Example:<br/><pre>peers:<br/>  - peer-address: 10.220.0.2<br/>    peer-asn: 65000<br/>    my-asn: 65002</pre> | |
```

> | bgp-communities | (optional) BGP community aliases. Instead of using hard to read BGP community numbers in address pool advertisement configurations, you can define alias names here and use those elsewhere in the configuration.<br/>Example:<br/><pre>bgp-communities:<br/>  no-export: 65535:65281</pre>||

Reviewer: ```suggestion
| bgp-communities | (optional) BGP community aliases. Instead of using hard-to-read BGP community numbers in address pool advertisement configurations, you can define alias names here and use those elsewhere in the configuration.<br/>Example:<br/><pre>bgp-communities:<br/>  no-export: 65535:65281</pre>||
```

| **address-pool** |||

> |name | A name for the address pool. Services can request allocation from a specific address pool using this name, by listing this name under the 'metallb.universe.tf/address-pool' annotation||

Reviewer: ```suggestion
|name | A name for the address pool. Services can request allocation from a specific address pool using this name, by listing this name under the `metallb.universe.tf/address-pool` annotation||
```

|protocol| Protocol can be used to select how the announcement is done. Supported values are bgp and layer2.||
|addresses| A list of IP address ranges over which MetalLB has authority. You can list multiple ranges in a single pool, they will all share the same settings. Each range can be either a CIDR prefix, or an explicit start-end range of IPs.<br/>Examples: <pre>addresses:<br/>  - 198.51.100.0/24<br/>  - 192.168.0.150-192.168.0.200</pre>||
|avoid-buggy-ips| (optional) If true, MetalLB will not allocate any address that ends in .0 or .255. Some old, buggy consumer devices mistakenly block traffic to such addresses under the guise of smurf protection. Such devices have become fairly rare, but the option is here if you encounter serving issues | false |
|auto-assign| (optional) If false, MetalLB will not automatically allocate any address in this pool. Addresses can still explicitly be requested via loadBalancerIP or the address-pool annotation | true |

> |bgp-advertisements[]| (optional) A list of **bgp-advertisement** to make, when protocol=bgp. Each address that gets assigned out of this pool will turn into this many advertisements. For most simple setups, you'll probably just want one. The default value for this field is a single advertisement with all parameters set to their respective defaults | All Default |

Reviewer: ```suggestion
|bgp-advertisements[]| (optional) A list of **bgp-advertisements** to make, when protocol=bgp. Each address that gets assigned out of this pool will turn into this many advertisements. For most simple setups, you'll probably just want one. The default value for this field is a single advertisement with all parameters set to their respective defaults. | All Default |
```

| **peer** |||
| peer-address | The target IP address for the BGP session ||
| peer-asn | The BGP AS number that MetalLB expects to see advertised by the router ||
| my-asn | The BGP AS number that MetalLB should speak as ||
| peer-port | (optional) the TCP port to talk to | 179 |
| source-address | (optional) The source IP address to use when establishing the BGP session. The address must be configured on a local network interface ||

> | hold-time | (optional) The proposed value of the BGP Hold Time timer. Refer to BGP reference material to understand what setting this implies ||

Reviewer: ```suggestion
| hold-time | (optional) The proposed value of the BGP Hold Time timer. Refer to BGP reference material to understand what setting this implies. ||
```

| **bgp-advertisement** |||
|aggregation-length| (optional) How much you want to aggregate up the IP address before advertising. For example, advertising 1.2.3.4 with aggregation-length=24 would end up advertising 1.2.3.0/24. For the majority of setups, you'll want to keep this at the default of 32, which advertises the entire IP address unmodified | 32 |
|aggregation-length-v6| (optional) How much you want to aggregate up the IPv6 address before advertising. For example, advertising 2001:0db8:85a3:0000:0000:8a2e:0370:7334 with aggregation-length-v6=64 would end up advertising 2001:0db8:85a3:0000:0000:0000:0000:0000/64. For the majority of setups, you'll want to keep this at the default of 128, which advertises the entire IP address unmodified|128|
|localpref| (optional) The value of the BGP "local preference" attribute for this advertisement. Only used with IBGP peers, i.e. peers where peer-asn is the same as my-asn. ||

> |communities[]| (optional) BGP communities to attach to this advertisement. Communities are given in the standard two-part form <asn>:<community number>. You can also use alias names ||

Reviewer: ```suggestion
|communities[]| (optional) BGP communities to attach to this advertisement. Communities are given in the standard two-part form \<asn\>:\<community number\>. You can also use alias names ||
```
It sees the greater-than and less-than signs as special characters. Check if just adding backslashes before each of those characters works.
